### PR TITLE
feat(*): init optional variable support - I145

### LINKED ISSUE #145: Optional Variables in Conditionals
# Feature Request 🛍️
Support the use of variables inside of if conditional blocks

## Context
Related to https://github.com/accordproject/cicero/issues/549

### PR DESCRIPTION
**Pull request by irmerk**
# Closes #145
Support `optional` variable types - conditionals with variables in them

### Changes
- Used very similar code to `conditional` variable support

### Flags
- Did need to repeat code from `Conditional`, DRY'd it as much as I could
- `Node.string(node)` can replace [`childReducer`](https://github.com/accordproject/web-components/blob/master/packages/ui-contract-editor/src/lib/components/actions.js#L21) from https://github.com/accordproject/web-components/pull/117
- Spacing (margin/padding) for variable inside optional could use work

### Related Issues
- https://github.com/accordproject/cicero/issues/549

### Author Checklist
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [x] Merging to `master` from `fork:branchname`

### CODE REVIEW

**Comment by Michael-Grover** [comment]
is it feasible to have the optional variable look nested within the conditional variable like this? 

image

Was thinking about editing the optional variable too. As a user I would expect if I click on the optional variable I could edit it (this would hide the tooltip that says "hide text"), but if I click outside of the optional variable but still within the conditional variable, then the text would be hidden.

**Comment by irmerk** [comment]
@Michael-Grover in the new deployment, the `border` of the `variable` inside the `optional` can only be as short in height as the `line-height`, so you may want to play around with that attribute to see if you'd want to change it.


> This looks great! However, we have the issue of "How do we edit the optional variable?" Basically, the comment I had here: [accordproject/models#108 (comment)](https://github.com/accordproject/models/pull/108#discussion_r450388900). Will that be handled in a separate PR?

This seems _very_ tricky. I think this would be best approached via the planned [pop-up editors](https://github.com/accordproject/web-components/issues/73). So for this PR, only rendering will be possible and not editing of the `variable` within an `optional`.

File: packages/ui-contract-editor/src/lib/styles.css [hunk 1]
```diff
@@ -6,17 +6,41 @@ a {
...
 
-.conditional {
-    cursor: pointer;
+.variable,
+.conditional,
+.optional {
+    border-radius: 2px;
+    background-color: #fff;
+    color: #000;
+    padding: 0 3px 1px 3px;
```
Comment (DianaLease) [inline packages/ui-contract-editor/src/lib/styles.css:15]: Did we want to change this to have the same top & bottom values? I think we were experimenting with `1px 3px` on our call.

Comment (irmerk) [reply packages/ui-contract-editor/src/lib/styles.css:15]: Ah yes!! 🙏 good catch

**Comment by jeromesimeon** [comment]
> @Michael-Grover in the new deployment, the `border` of the `variable` inside the `optional` can only be as short in height as the `line-height`, so you may want to play around with that attribute to see if you'd want to change it.
> 
> > This looks great! However, we have the issue of "How do we edit the optional variable?" Basically, the comment I had here: [accordproject/models#108 (comment)](https://github.com/accordproject/models/pull/108#discussion_r450388900). Will that be handled in a separate PR?
> 
> This seems _very_ tricky. I think this would be best approached via the planned [pop-up editors](https://github.com/accordproject/web-components/issues/73). So for this PR, only rendering will be possible and not editing of the `variable` within an `optional`.

I might be wrong but I'm not sure the pop up helps. My impression is that either way we have to help the editor figure out where it clicks and handle precedence (e.g., variable is inside an optional which is inside a clause which is inside a contract). once we have a way to clearly disambiguate it can be editing or pop up or anything else we want depending on the context?

**Comment by irmerk** [comment]
Addressed the first two points @Michael-Grover. For the overlap, not sure what you're looking for.

Note that in `packages/ui-contract-editor/src/lib/styles.css`, there are styling rules for `optional`+`conditional`+`variable`, `optional`+`conditional`, and `variable` WITHIN an `optional`. Not sure how you got your last screenshot without overlap.

**Pull request merged**